lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -2553,16 +2575,22 @@ Value *PatchInOutImportExport::patchFsBuiltInInputImport(Type *inputTy, unsigned
     const auto prevStage = m_pipelineState->getPrevShaderStage(ShaderStageFragment);
 
     if (prevStage == ShaderStageMesh) {
-      llvm_unreachable("Not implemented!");
+      assert(inOutUsage.perPrimitiveBuiltInInputLocMap.count(builtInId) > 0);
+      loc = inOutUsage.perPrimitiveBuiltInInputLocMap[builtInId];
     } else {
       assert(inOutUsage.builtInInputLocMap.count(builtInId) > 0);
       loc = inOutUsage.builtInInputLocMap[builtInId];
     }
 
     // Emulation for "in int gl_PrimitiveID" or "in int gl_Layer" or "in int gl_ViewportIndex"
     // or "in int gl_ViewIndex"
-    input = patchFsGenericInputImport(inputTy, loc, nullptr, nullptr, nullptr, InOutInfo::InterpModeFlat,
-                                      InOutInfo::InterpLocCenter, false, insertPos);
+    bool isPerPrimitive = false;
+    if (m_pipelineState->getPrevShaderStage(ShaderStageFragment) == ShaderStageMesh) {
+      // NOTE: If the previous shader stage is mesh shader, those built-ins are exported via primitive attributes.
+      isPerPrimitive = true;
+    }
```

Comment:
You could set `isPerPrimitive` in the if/else above.

Reply:
OK. Will be updated.